x/rollapp/types/rollapp.go

```diff
@@ -144,76 +151,77 @@ func validateInitialSequencer(initialSequencer string) error {
 func validateBech32Prefix(prefix string) error {
 	bechAddr, err := sdk.Bech32ifyAddressBytes(prefix, sample.Acc())
 	if err != nil {
-		return errorsmod.Wrap(err, "bech32ify addr bytes")
+		return err
 	}
 
 	bAddr, err := sdk.GetFromBech32(bechAddr, prefix)
 	if err != nil {
-		return errorsmod.Wrap(err, "get from bech 32")
+		return err
 	}
 
 	if err = sdk.VerifyAddressFormat(bAddr); err != nil {
-		return errorsmod.Wrap(err, "verify addr format")
+		return err
 	}
 	return nil
 }
 
 func (dm DenomMetadata) Validate() error {
 	if l := len(dm.Base); l == 0 || l > maxDenomBaseLength {
-		return errorsmod.Wrap(gerrc.ErrInvalidArgument, "base denom")
+		return fmt.Errorf("base denom")
 	}
 
 	if l := len(dm.Display); l == 0 || l > maxDenomDisplayLength {
```

Comment:
we're missing validations here of the sdk (to prevent from validating again after):

```
if err := sdk.ValidateDenom(m.Base); err != nil {
		return fmt.Errorf("invalid metadata base denom: %w", err)
	}

	if err := sdk.ValidateDenom(m.Display); err != nil {
		return fmt.Errorf("invalid metadata display denom: %w", err)
	}
```